core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala

```diff
@@ -64,8 +64,8 @@ class JdbcRDD[T: ClassTag](
     // bounds are inclusive, hence the + 1 here and - 1 on end
     val length = 1 + upperBound - lowerBound
     (0 until numPartitions).map(i => {
-      val start = lowerBound + ((i * length) / numPartitions).toLong
-      val end = lowerBound + (((i + 1) * length) / numPartitions).toLong - 1
+      val start = lowerBound + ((BigDecimal(i) * length) / numPartitions).toLong
```

Comment:
I think there is a problem here where 
length = 1 + upperBound - lowerBound 
This can over flow as well, if lowerBound = 0 and upperBound = Long.MAX_VALUE
I should fix this, actually make the change smaller